### PR TITLE
Add notifier for Smtp

### PR DESCRIPTION
**Pull request by utkarsharma2**
Send email via SMTP notifier.

Example usage -  
```
BashOperator(
            task_id="mytask",
            on_failure_callback=[
                send_smtp_notification(
                    mail_from="someone@mail.com",
                    to="someone@mail.com",
                    subject="[Error] The Task {{ ti.task_id }} failed",
                    html_content="debug logs",
                )
            ],
            bash_command="fail",
        )
```

### CODE REVIEW

File: airflow/providers/smtp/notifications/smtp.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        "cc",
+        "bcc",
+        "mime_subtype",
+        "mime_charset",
+        "custom_headers",
+    )
+
+    def __init__(
+        self,
+        mail_from: str | None,
```
Comment (jedcunningham) [inline airflow/providers/smtp/notifications/smtp.py]:
```suggestion
        *,
        from_email: str | None,
```

Let's make these kwarg only.

Let's also use the name from the operator and hook, `from_email`.

File: airflow/providers/smtp/notifications/smtp.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    def hook(self) -> SmtpHook:
+        """Smtp Events Hook"""
+        return SmtpHook(smtp_conn_id=self.smtp_conn_id)
+
+    def notify(self, context):
+        """Send a email via smtp server"""
+        with self.hook as smtp:
+            smtp.send_email_smtp(
+                smtp_conn_id=self.smtp_conn_id,
+                mail_from=self.mail_from,
```
Comment (jedcunningham) [inline airflow/providers/smtp/notifications/smtp.py]:
```suggestion
                mail_from=self.mail_from,
```

This should be `from_email`. Make sure you test on a connection without a [from_email set](https://github.com/apache/airflow/blob/c5597d1fabe5d8f3a170885f6640344d93bf64bf/airflow/providers/smtp/hooks/smtp.py#L327).

Comment (jedcunningham) [reply airflow/providers/smtp/notifications/smtp.py]: I have no idea why `send_email_smtp` accepts extra kwargs. It just hid this issue :(

Comment (utkarsharma2) [reply airflow/providers/smtp/notifications/smtp.py]: Ya, that hid the issue, made the changes suggested, and also made sure the `from_email` is picked when passed in `send_smtp_notification`.

File: airflow/providers/smtp/notifications/smtp.py (hunk 1 above)
Comment (utkarsharma2) [reply airflow/providers/smtp/notifications/smtp.py]: Done.

**Comment by utkarsharma2** [comment]
The static checks failing are unrelated to this PR, please check the slack thread - https://apache-airflow.slack.com/archives/CCPRP7943/p1685011999082929

**Review by jedcunningham — APPROVED** [review]

File: airflow/providers/smtp/notifications/smtp.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                files=self.files,
+                cc=self.cc,
+                bcc=self.bcc,
+                mime_subtype=self.mime_subtype,
+                mime_charset=self.mime_charset,
+                custom_headers=self.custom_headers,
+            )
+
+
+send_smtp_notification = SmtpNotifier
```
Comment (dstandish) [inline airflow/providers/smtp/notifications/smtp.py:106]: why do we do this?

Comment (utkarsharma2) [reply airflow/providers/smtp/notifications/smtp.py:106]: Better naming, I guess. I was following convention on this PR - https://github.com/apache/airflow/pull/28569/files

File: airflow/providers/smtp/notifications/smtp.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Any, Iterable
+
+from airflow.compat.functools import cached_property
```
Comment (ephraimbuddy) [inline airflow/providers/smtp/notifications/smtp.py]: We can import this from python directly now since 3.7 has been removed

**Review by ephraimbuddy — APPROVED** [review]

**Pull request merged**